src/components/ha-date-input.ts

```diff
@@ -0,0 +1,124 @@
+import {
+  html,
+  css,
+  LitElement,
+  TemplateResult,
+  property,
+  customElement,
+} from "lit-element";
+
+import { PaperInputElement } from "@polymer/paper-input/paper-input";
+
+@customElement("ha-date-input")
+class HaDateInput extends LitElement {
+  @property() public year?: string;
+  @property() public month?: string;
+  @property() public day?: string;
+  @property() public disabled?: boolean = false;
```

Comment:
- Usually no need to set TS type if you set a default value.
- If it's boolean, set the `property` type, so existence of the attribute, even with no value, means it's `true`

```suggestion
  @property({type: Boolean}) public disabled = false;
```